### DISCUSSION TITLE
Access Token Expiration

### DISCUSSION THREAD

**Discussion by mark3grahams** [Snow-Shell/servicenow-powershell#224] [General]
Hi,

I am new to using this module.

After a while of use, I start getting "(401) Unauthorized."

I create a new session using "New-ServiceNowSession" everything is fine for the Access Token life of the new token.

Is this normal or should the module be requesting a new Access Token using the RefreshToken? Or is there some other way to do this that I am missing?

**Comment by gdbarron** [comment]
Hi @mark3grahams.  There is currently no auto-renew functionality for tokens when they expire.  You could either check for this in code prior to subsequent calls or up the lifespan.  Feel free to submit an enhancement as well.  Out of curiosity, how long is the lifespan?

**Comment by mark3grahams** [comment]
The default seems to be 30 minutes, which isn't a big deal. I was more curious about if the refresh token was used and how.

**Reply by gdbarron** [reply]
The refresh token can be used, but the functionality needs to be added to the module. I've done this with others, but not this one. If it's in your wheelhouse, feel free to submit a pull request with the update.
